client/asset/dcr/dcr.go

```diff
@@ -885,6 +918,39 @@ func (dcr *ExchangeWallet) feeRate(confTarget uint64) (uint64, error) {
 	return 1 + uint64(atomsPerKB)/1000, nil // dcrPerKB * 1e8 / 1e3
 }
 
+// externalFeeEstimator gets the fee rate from the external API
+func externalFeeEstimator(ctx context.Context, net dex.Network, nb uint64) (float64, error) {
+	var url string
+	if net == dex.Testnet {
+		url = testnetExternalApiUrl
+	} else {
+		url = externalApiUrl
+	}
+	url = url + "/utils/estimatefee?nbBlocks=" + strconv.FormatUint(nb, 10)
+	ctx, cancel := context.WithTimeout(ctx, 4*time.Second)
+	defer cancel()
+	r, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return 0, err
+	}
+	httpResponse, err := client.Do(r)
+	if err != nil {
+		return 0, err
+	}
+	c := make(map[uint64]float64)
+	reader := io.LimitReader(httpResponse.Body, 1<<14)
+	err = json.NewDecoder(reader).Decode(&c)
+	httpResponse.Body.Close()
+	if err != nil {
+		return 0, err
+	}
+	estimatedFeeRate, ok := c[nb]
+	if !ok {
+		return 0, errors.New("no fee rate for requested number of blocks")
+	}
+	return estimatedFeeRate, nil
+}
+
```

Comment:
A few lines down there's still a `if dcr.wallet.SpvMode() {` that skips calling `feeRate` where the external API would be called.  Thus we're still falling back in a lot of cases.

Reply:
lol I completely missed it. Removed it now, thanks for catching.